homeassistant/components/sensor/seventeentrack.py

```diff
@@ -139,6 +142,20 @@ def unit_of_measurement(self):
         """Update the sensor."""
         await self._data.async_update()
 
+        package_data = []
+        for package in [
+                p for p in self._data.packages if p.status == self._status
```

Comment:
This will make it two loops. We only need one, right?
```py
for package in self._data.packages:
    if package.status != self._status:
        continue
    package_data.append(...)
```